services/packages/client/client.go

```diff
@@ -348,8 +349,119 @@ func (l *listCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interfac
 		}
 		fmt.Fprint(state.Out[r.Index], "Installed Packages\n")
 		for _, pkg := range r.Resp.Packages {
-			// Print the package name, version and repo with some reasonable spacing.
-			fmt.Fprintf(state.Out[r.Index], "%40s %16s %32s\n", pkg.Name, pkg.Version, pkg.Repo)
+			// Print the package name.arch, [epoch]:version-release and repo with some reasonable spacing.
+			na := fmt.Sprintf("%s.%s", pkg.Name, pkg.Architecture)
+			evr := fmt.Sprintf("%s-%s", pkg.Version, pkg.Release)
+			if pkg.Epoch != 0 {
+				evr = fmt.Sprintf("%d:%s", pkg.Epoch, evr)
+			}
+			fmt.Fprintf(state.Out[r.Index], "%40s %16s %32s\n", na, evr, pkg.Repo)
+		}
+	}
+	return retCode
+}
+
+type searchCmd struct {
+	packageSystem string
+	name          string
+	installed     bool
+	latest        bool
+}
+
+func (*searchCmd) Name() string     { return "search" }
+func (*searchCmd) Synopsis() string { return "Search NEVRA of a package" }
+func (*searchCmd) Usage() string {
+	return `search --name=X [--installed] [--latest] [--package_system=P]:
+  Search the current installed packages or the latest packages to be updated on the remote machine. The packages will be displayed in NEVRA version.
+`
+}
+
+func (l *searchCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&l.packageSystem, "package-system", "YUM", fmt.Sprintf("Package system to use(one of: [%s])", strings.Join(shortPackageSystemNames(), ",")))
+	f.StringVar(&l.name, "name", "", "Name of package to search")
+	f.BoolVar(&l.installed, "installed", false, "If true print out installed NEVRA of the package")
+	f.BoolVar(&l.latest, "latest", false, "If true print out latest NEVRA of the package")
```

Comment:
I'd set both default to true, otherwise we have to specify a flag on every run.

Reply:
By default(don't specify any flags), these two flags will be set to true here: https://github.com/Snowflake-Labs/sansshell/pull/271/files#diff-58624590b1dc636ebdcbb5fddf74fa9c9a06d87765bd48593b98e66b2292197dR409 (I will remove the vars and use flags)